ckan/cli/cli.py

```diff
@@ -39,22 +39,29 @@
 
 class CkanCommand(object):
 
-    def __init__(self, conf=None):
+    def __init__(self, conf=None, setup_logging=True):
         # Don't import `load_config` by itself, rather call it using
         # module so that it can be patched during tests
-        self.config = ckan_cli.load_config(conf)
+        self.config = ckan_cli.load_config(conf, setup_logging=setup_logging)
         self.app = make_app(self.config)
 
 
-def _init_ckan_config(ctx, param, value):
+def _init_ckan_config(ctx, param, config_filepath):
+    setup_logging = True
+    if len(sys.argv) > 1 and \
+            sys.argv[-2:] == [u'datastore', u'set-permissions']:
```

Comment:
if someone runs the command like `ckan datastore set-permissions -c my-config.ini` then this won't work.

Reply:
I'm not sure that is valid. Doesn't the param has to go before the subcommand?

Reply:
you're right, that doesn't work with the new click cli